### ISSUE TITLE
Charts deprecated

### ISSUE THREAD

**Issue by volker-raschek** [drone/charts#46]
I've seen that drone v2.0.0 has been released, but the chart version has since more than a year not been updated. Are the charts deprecated or orphaned?

Maybe an update of the helm charts are required to deploy drone v2.0 in a kubernetes cluster.

**Comment by ElArtista** [comment]
Seems like the whole repo is unmaintained, half a year since latest commit, no response in any of the recent PRs etc..

**Comment by tolgap** [comment]
@bradrydzewski can we expect updates to the chart for drone 2.0? Thank you very much 😄

**Comment by juliantellez** [comment]
happy to help this happen

**Comment by volker-raschek** [comment]
Hi folks, 
it takes too long time for me until someone updates or maintains the repository again. For this reason I have started to package drone via helm for myself. Maybe anyone is interessed to test and/or contribute to the helm chart for v2. 

[artifacthub](https://artifacthub.io/packages/helm/volker-raschek/drone)
[repository](https://github.com/volker-raschek/drone-charts)

Volker

**Comment by janishorsts** [comment]
Yes, it looks like this repo is a bit abandoned, but you can still use these charts to upgrade to the latest drone versions by specifying which image tags to pull.

These charts won't work with ingress enabled and kubectl v1.22.

```yaml
# drone chart values
image:
  tag: 2.4.0
```

```yaml
# drone runner kube chart values
image:
  tag: 1.0.0-rc.1
```

**Comment by Paragon1970** [comment]
Certainly does look like this repo has been abandoned, guess its due to the sale of DroneCI to Harness :hunch:, i dunno! 

#### Regarding ingress on v1.22+ 

Original template:
https://github.com/drone/charts/blob/master/charts/drone/templates/ingress.yaml

Edit the version condition something like this: (not tested)
```
{{- if semverCompare "<1.22-0" .Capabilities.KubeVersion.GitVersion -}}
apiVersion: networking.k8s.io/v1beta1
{{- else -}}
apiVersion: networking.k8s.io/v1
{{- end }}
```

Dont use:
```
annotation: 
      kubernetes.io/ingress.class: nginx-internal
```
instead employ:
```
spec:
    ingressClassName: "fooclassname"
```
Change following directives and if employing version conditions they would need to be added here also.

```
backend:
      serviceName: {{ $fullName }}
      servicePort: {{ $svcPort }}
```             
to:
```
backend:
      service:
            name: {{ $fullName }}
            port:
              number: {{ $svcPort }}
```

Alternativly one could employ a single version condition and have multiple template bodies.

```
{{- if semverCompare "<1.22-0" .Capabilities.KubeVersion.GitVersion -}}
apiVersion: networking.k8s.io/v1beta1
<ingress v1beta directive>
{{- else -}}
apiVersion: networking.k8s.io/v1
< ingress v1 template directives>
{{- end }}
```

Of course all the above implies you to either fork the repo to make changes 
or
Employ subchart with chart dependancies holding local self managed template for ingress.

**Comment by jimsheldon** [comment]
Apologies for how long it has taken to give this issue attention.

I have submitted PR #53 which brings the drone server chart up to 2.11.1, updates the ingress to be compatible with newer versions of Kubernetes, and updates liveness/readiness probes.

If anyone is able to try it from the `drone-chart-updates` branch and provide feedback, I would really appreciate it!

**Comment by jimsheldon** [comment]
PR #53 has been merged which has pushed version 0.2.0 of the drone chart with the changes mentioned above.

I will close out this issue. Please submit a new issue if this PR did not address a problem you are having.
